nifi-nar-bundles/nifi-cassandra-bundle/nifi-cassandra-processors/src/main/java/org/apache/nifi/processors/cassandra/PutCassandraRecord.java

```diff
@@ -67,6 +75,35 @@
             .required(true)
             .build();
 
+    static final PropertyDescriptor STATEMENT_TYPE = new PropertyDescriptor.Builder()
+            .name("put-cassandra-record-statement-type")
+            .displayName("Statement Type")
+            .description("Specifies the type of CQL Statement to generate.")
+            .required(true)
+            .defaultValue(INSERT_TYPE)
+            .allowableValues(UPDATE_TYPE, INSERT_TYPE)
+            .build();
+
+    static final PropertyDescriptor UPDATE_METHOD = new PropertyDescriptor.Builder()
+            .name("put-cassandra-record-update-method")
+            .displayName("Update Method")
+            .description("Specifies the method to use to SET the values.")
```

Comment:
This should include in the description that the property is only used if the Statement Type is Update, and ignored otherwise.

Also, do you anticipate multiple flow files coming in, each needing a different Update Method? You can't use Expression Language with AllowableValues/dropdowns, but a common pattern is to have an AllowableValue whose display name is `Use cql.update.method attribute`, and if that is chosen, the `cql.update.method` attribute is expected to be set to the name of one of the available options. See PutDatabaseRecord for an example of this pattern. Same goes for the Statement Type, if you allow that to be set by an attribute, then you could send all flow files to one PutCassandraRecord instance, rather than having to route on type and then send to 2+ instances (one for INSERT, one for UPDATE, e.g.)

Reply:
Do you see other Statement Types being included someday, such as UPSERT, DELETE, etc.? I ask because I wonder if we need a separate property for Update Method or if we can include 3 Statement Types (Update Set, Update Increment, Update Decrement) and only use one property. Then if we add UPSERT, DELETE, etc. we don't have extra properties that only pertain to one of the Statement Types.

Either way is fine with me, just trying to design for the future so we don't run into compatibility or awkward configuration issues (too many properties, e.g.)

Reply:
Hey @mattyb149 I don't see many more statement types, as Cassandra doesn't really have a notion of upsert (an insert and update are already mostly the same, aside from COUNTER tables where insert doesn't work). The only thing I see in the future is a DELETE statement, for which the update methods would also not be applicable. On the other hand, this feels cleaner than having "Update Set, Update Increment,..". I'm fine either way so if fewer properties is preferred then I can get rid of the update method.

Depending on your answer on the above, I'll add the cql.update.method attribute and/or cql.statement.type attribute as well.

Reply:
I've modified Statement Type, Update Method as well as Batch Statement Type (in case of counter tables can only be counter) to also take their value from an attribute.